src/createStreamingManager.ts

```diff
@@ -37,57 +37,58 @@ function mapConnectionState(state: RTCIceConnectionState): ConnectionState {
-                        let videoStatsReport;
-
-                        isPlaying = currBytesReceived - lastBytesReceived > 0;
-                        isPlayingFalseNumIntervals = isPlaying ? 0 : ++isPlayingFalseNumIntervals;
-
-                        if (prevPlaying !== isPlaying) {
-                            if (isPlaying) {
-                                videoStatsStartIndex = videoStats.length;
-                                onVideoStateChange?.(StreamingState.Start, videoStatsReport);
-                            } else {
-                                const stats = videoStats.slice(videoStatsStartIndex);
-                                const previousStats =
-                                    videoStatsStartIndex === 0 ? undefined : videoStats[videoStatsStartIndex - 1];
-                                videoStatsReport = createVideoStatsReport(stats, previousStats);
-                                videoStatsReport = videoStatsReport
-                                    .sort((a, b) => b.packetsLost - a.packetsLost)
-                                    .slice(0, 5);
-                            }
-                        }
-
-
-                        if (!isPlaying && isPlayingFalseNumIntervals === 1) {
-                            analytics.track('agent-video', { event: 'stats', rtc: videoStatsReport });
-                        }
-
-                        if (!isPlaying && isPlayingFalseNumIntervals === 2) {
-                            onVideoStateChange?.(StreamingState.Stop, videoStatsReport);
-                        }
-                    }
-                    videoStats.push(report);
+function createVideoStatsAnalyzer() {
+    let lastBytesReceived = 0;
+
+    return (stats: RTCStatsReport) => {
+        stats.forEach(report => {
+            if (report.type === 'inbound-rtp' && report.kind === 'video') {
+                if (report) {
+                    const currBytesReceived = report.bytesReceived;
+                    const isReceiving = currBytesReceived - lastBytesReceived > 0;
+
+                    lastBytesReceived = currBytesReceived
+
+                    return isReceiving
                 }
-            });
+            }
         });
-    }, 500);
+
+        return false;
+    };
+}
+
+function pollStats(peerConnection: RTCPeerConnection, onVideoStateChange, analytics) {
+    const interval = 100;
+    const notReceivingIntervalsThreshold = Math.max(Math.ceil(1000 / interval), 1);
+
+    let notReceivingNumIntervals = 0;
+    let isStreaming = false;
+
+    const isReceivingBytes = createVideoStatsAnalyzer();
+
+    return setInterval(async () => {
+        const stats = await peerConnection.getStats();
+
+        if (isReceivingBytes(stats)) {
+            notReceivingNumIntervals = 0;
+
+            if (!isStreaming) {
+                onVideoStateChange?.(StreamingState.Start);
+
+                isStreaming = true;
+            }
+
+        } else {
+            notReceivingNumIntervals++;
```

Comment:
you always increase it no?

Reply:
as long as not receiving no ? 
I zero it overhere:
if (isReceivingBytes(stats)) {
            notReceivingNumIntervals = 0;

            if (!isStreaming) {
                onVideoStateChange?.(StreamingState.Start);

                isStreaming = true;
            }

        }

Reply:
done